rainbow-me/rainbow
# E2E improvements

Fixes APP-####

## What changed (plus any additional context for devs)
- Replace Hardhat in favor of Anvil
- Replace usage of `IS_TESTING` in favor of `IS_TEST`
- Fix the detox config to actually run the correct tests for serial & parallel. This wasn't working and tests were running on the wrong queue.
- Update the script with more efficient retry logic so it only runs what failed instead of the entire suite
- Run anvil separately so we can see the logs that are actually useful during the job execution

## Screen recordings / screenshots

Before ~ 21min

<img width="739" alt="Screenshot 2024-12-21 at 8 02 51 PM" src="https://github.com/user-attachments/assets/ba1a06c0-9212-48bd-baa0-e43401ba2ac5" />

Now ~ 15min
<img width="739" alt="Screenshot 2024-12-21 at 8 03 00 PM" src="https://github.com/user-attachments/assets/c5537797-f055-4d50-845a-2634ce89be30" />


## What to test
e2e should pass
